src/modules/fancyzones/lib/JsonHelpers.h

```diff
@@ -159,12 +160,27 @@ namespace JSONHelpers
 
     class FancyZonesData
     {
+        mutable std::recursive_mutex dataLock;
+
     public:
         FancyZonesData();
 
-        const std::wstring& GetPersistFancyZonesJSONPath() const;
+        inline const std::wstring& GetPersistFancyZonesJSONPath() const
+        {
+            return jsonFilePath;
+        }
         json::JsonObject GetPersistFancyZonesJSON();
 
+        std::optional<DeviceInfoData> FindDeviceInfo(const std::wstring& zoneWindowId) const;
+
+        std::optional<CustomZoneSetData> FindCustomZoneSet(const std::wstring& guuid) const;
+
+        inline const std::wstring GetActiveDeviceId() const
+        {
+            return activeDeviceId;
+        }
+
+#if defined(UNIT_TESTS)
```

Comment:
I don't think so, the locks are needed only because of the PT events being supplied from a separate thread. See https://github.com/microsoft/PowerToys/issues/961.